azext_iot/custom.py

```diff
@@ -800,3 +815,279 @@ def iot_device_upload_file(client, device_id, hub_name, file_path, content_type,
+        
+        enrollment_record = m_sdk.device_enrollment_group.get(enrollment_id, API_VERSION)
+        if not etag == enrollment_record['etag'].replace('"', ''):
+            raise LookupError("enrollment etag not found.")
+
+        if iot_hub_host_name:
+            enrollment_record['iotHubHostName'] = iot_hub_host_name
+        if provisioning_status:
+            enrollment_record['provisioningStatus'] = provisioning_status
+ 
+        enrollment_record['initialTwin'] = _get_updated_inital_twin(enrollment_record, 
+                                                                  initial_twin_tags, 
+                                                                  initial_twin_properties)
+        
+        enrollment_record['attestation'] = _get_attestation_with_x509_signing_cert(certificate_path)
+        
+        return m_sdk.device_enrollment_group.create_or_update(enrollment_id, enrollment_record, API_VERSION, etag)
+    except errors.ErrorDetailsException as e:
+        raise CLIError(e)
+
+def iot_dps_device_enrollment_group_delete(client, enrollment_id, dps_name, resource_group_name):
+    target = get_iot_dps_connection_string(client, dps_name, resource_group_name)
+    try:
+        m_sdk, errors = _bind_sdk(target, SdkType.dps_sdk)
+        return m_sdk.device_enrollment_group.delete(enrollment_id, API_VERSION)
+    except errors.ErrorDetailsException as e:
+        raise CLIError(e)
+
+# DPS Registration
+def iot_dps_registration_list(client, dps_name, resource_group_name, enrollment_id):
+    target = get_iot_dps_connection_string(client, dps_name, resource_group_name)
+    try:
+        m_sdk, errors = _bind_sdk(target, SdkType.dps_sdk)
+        return m_sdk.registration_status.query_registration_state(enrollment_id, API_VERSION)
+    except errors.ErrorDetailsException as e:
+        raise CLIError(e)
+
+def iot_dps_registration_get(client, dps_name, resource_group_name, registration_id):
+    target = get_iot_dps_connection_string(client, dps_name, resource_group_name)
+    try:
+        m_sdk, errors = _bind_sdk(target, SdkType.dps_sdk)
+        return m_sdk.registration_status.get_registration_state(registration_id, API_VERSION)
+    except errors.ErrorDetailsException as e:
+        raise CLIError(e)
+
+def iot_dps_registration_delete(client, dps_name, resource_group_name, registration_id):
+    target = get_iot_dps_connection_string(client, dps_name, resource_group_name)
+    try:
+        m_sdk, errors = _bind_sdk(target, SdkType.dps_sdk)
+        return m_sdk.registration_status.delete_registration_state(registration_id, API_VERSION)
+    except errors.ErrorDetailsException as e:
+        raise CLIError(e)
+
+def _get_initial_twin(initial_twin_tags = None, initial_twin_properties = None):
+    import ast
+    if not initial_twin_tags and not initial_twin_properties:
+        return None
+    if initial_twin_tags:
+        initial_twin_tags = evaluate_literal(str(initial_twin_tags), dict)
```

Comment:
Please use the existing utility evaluate_literal, it will return None if the input is not parsable.

Reply:
Ah i thought i ready did, but i forgot to remove the import ast at the top